pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -1173,13 +1230,17 @@ public void setProperty(PGProperty property, String value) {
     }
     switch (property) {
       case PG_HOST:
-        serverName = value;
+        serverNames = value.split(",");
         break;
       case PG_PORT:
-        try {
-          portNumber = Integer.parseInt(value);
-        } catch (NumberFormatException e) {
-          portNumber = 0;
+        String[] ps = value.split(",");
```

Comment:
This should delegate to `setPortNumbers(...)` rather than direct assignment. That way it handles the empty array situation.